From 5d9547c477211df1ea198f1cc05e0edb3ec5083d Mon Sep 17 00:00:00 2001
From: port <p@p>
Date: Tue, 20 Oct 2026 12:39:14 +0000
Subject: [PATCH] configure: Add the DirectAudio (Android AAudio) driver

Integration points from The412Banner/directaudio INTEGRATION.md. The driver sources are copied into dlls/winedirectaudio.drv at build time and configure is regenerated.
---
 configure.ac | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index f63669d..18a0b65 100644
--- a/configure.ac
+++ b/configure.ac
@@ -27,6 +27,7 @@ AC_ARG_ENABLE(silent-rules, AS_HELP_STRING([--enable-silent-rules],[use silent b
 AC_ARG_ENABLE(werror, AS_HELP_STRING([--enable-werror],[treat compilation warnings as errors]))
 
 AC_ARG_WITH(alsa,      AS_HELP_STRING([--without-alsa],[do not use the Alsa sound support]))
+AC_ARG_WITH(aaudio,    AS_HELP_STRING([--without-aaudio],[do not use the Android AAudio sound support]))
 AC_ARG_WITH(capi,      AS_HELP_STRING([--without-capi],[do not use CAPI (ISDN support)]))
 AC_ARG_WITH(coreaudio, AS_HELP_STRING([--without-coreaudio],[do not use the CoreAudio sound support]))
 AC_ARG_WITH(cups,      AS_HELP_STRING([--without-cups],[do not use CUPS]))
@@ -1072,6 +1073,15 @@ case $host_os in
 esac
 
 enable_winecoreaudio_drv=${enable_winecoreaudio_drv:-no}
+dnl **** Check for AAudio (Android native audio) for the DirectAudio driver ****
+if test "x$with_aaudio" != "xno"
+then
+    AC_CHECK_HEADER([aaudio/AAudio.h],
+        [AC_SUBST(AAUDIO_LIBS,"-laaudio")
+         enable_winedirectaudio_drv=${enable_winedirectaudio_drv:-yes}])
+fi
+
+enable_winedirectaudio_drv=${enable_winedirectaudio_drv:-no}
 enable_wineandroid_drv=${enable_wineandroid_drv:-no}
 enable_winemac_drv=${enable_winemac_drv:-no}
 
@@ -1921,8 +1931,8 @@ WINE_NOTICE_WITH(netapi,[test "x$ac_cv_lib_soname_netapi" = "x"],
 
 
 dnl **** Check for any sound system ****
-if test "x$enable_winealsa_drv$enable_winecoreaudio_drv$enable_winepulse_drv$enable_wineoss_drv$enable_wineandroid_drv" = xnonononono -a \
-        "x$with_alsa$with_coreaudio$with_oss$with_pulse" != xnononono
+if test "x$enable_winealsa_drv$enable_winecoreaudio_drv$enable_winedirectaudio_drv$enable_winepulse_drv$enable_wineoss_drv$enable_wineandroid_drv" = xnononononono -a \
+        "x$with_aaudio$with_alsa$with_coreaudio$with_oss$with_pulse" != xnonononono
 then
     WINE_WARNING([No sound system was found. Windows applications will be silent.])
 fi
@@ -3352,6 +3362,7 @@ WINE_CONFIG_MAKEFILE(dlls/windowscodecs/tests)
 WINE_CONFIG_MAKEFILE(dlls/windowscodecsext)
 WINE_CONFIG_MAKEFILE(dlls/windowscodecsext/tests)
 WINE_CONFIG_MAKEFILE(dlls/winealsa.drv)
+WINE_CONFIG_MAKEFILE(dlls/winedirectaudio.drv)
 WINE_CONFIG_MAKEFILE(dlls/wineandroid.drv)
 WINE_CONFIG_MAKEFILE(dlls/winebth.sys)
 WINE_CONFIG_MAKEFILE(dlls/winebus.sys)
